src/components/Karma/utils.ts

```diff
@@ -0,0 +1,19 @@
+export const renderCircleColor = (karma: number) => {
+  switch (true) {
+    case karma > 87.5:
+      return ['#50BA57', '#C7EECA', '#50BA57'];
+    case karma > 75:
+      return ['#70BF21', '#B3E57F', '#70BF21'];
+    case karma > 62.5:
+      return ['#A0C527', '#D1EA82', '#A0C527'];
+    case karma > 50:
+      return ['#D3D20C', '#F3F370', '#D3D20C'];
+    case karma > 37.5:
+      return ['#E4B92E', '#FFE591', '#E4B92E'];
+    case karma > 25:
+      return ['#F4833C', '#F9BE9A', '#F4833C'];
+    case karma > 12.5:
+    default:
+      return ['#DF001F', '#F5B0B9', '#DF001F'];
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `renderCircleColor` function uses a switch statement with `true` as the condition and checks if karma is greater than certain values in each case. This approach works but it's not very intuitive or conventional. A more common pattern would be to use if/else if statements. Here's how you could refactor this:

```diff
-  switch (true) {
-    case karma > 87.5:
-      return ['#50BA57', '#C7EECA', '#50BA57'];
-    case karma > 75:
-      return ['#70BF21', '#B3E57F', '#70BF21'];
-    case karma > 62.5:
-      return ['#A0C527', '#D1EA82', '#A0C527'];
-    case karma > 50:
-      return ['#D3D20C', '#F3F370', '#D3D20C'];
-    case karma > 37.5:
-      return ['#E4B92E', '#FFE591', '#E4B92E'];
-    case karma > 25:
-      return ['#F4833C', '#F9BE9A', '#F4833C'];
-    case karma > 12.5:
-    default:
-      return ['#DF001F', '#F5B0B9', '#DF001F'];
-  }
+  if (karma > 87.5) {
+    return ['#50BA57', '#C7EECA', '#50BA57'];
+  } else if (karma > 75) {
+    return ['#70BF21', '#B3E57F', '#70BF21'];
+  } else if (karma > 62.5) {
+    return ['#A0C527', '#D1EA82', '#A0C527'];
+  } else if (karma > 50) {
+    return ['#D3D20C', '#F3F370', '#D3D20C'];
+  } else if (karma > 37.5) {
+    return ['#E4B92E', '#FFE591', '#E4B92E'];
+  } else if (karma > 25) {
+    return ['#F4833C', '#F9BE9A', '#F4833C'];
+  } else {
+    return ['#DF001F', '#F5B0B9', '#DF001F'];
+  }
```
This way, the code is more readable and follows a more common pattern.


<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
iya mas ucup, karna ini berjangka ( ada range ) , pake if else aja, cangan switch case.